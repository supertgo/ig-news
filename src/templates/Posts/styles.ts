import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 112rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.medium};
  `}
`;

export const Content = styled.div`
  max-width: 72rem;
  margin: 8rem auto 0;
`;

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;

    & + a {
      margin-top: ${theme.spacings.medium};
      padding-top: ${theme.spacings.medium};
      border-top: 1px solid #323238;
    }
  `}
`;

export const Time = styled.time`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    display: flex;
    align-items: center;
    color: ${theme.colors.text};
  `}
`;

export const Title = styled.strong`
  ${({ theme }) => css`
    display: block;
    font-size: 2.4rem;
    margin-top: ${theme.spacings.medium};
    line-height: ${theme.spacings.medium};
    transition: color 0.2s;

    &:hover {
      color: ${theme.colors.yellow};
    }
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    margin-top: ${theme.spacings.xxsmall};
    line-height: 2.6rem;
  `}
`;
