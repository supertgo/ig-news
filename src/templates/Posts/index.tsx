import Head from 'next/head';
import Base from 'templates/Base';

import * as S from './styles';

type Post = {
  slug: string;
  title: string;
  excerpt: string;
  updatedAt: string;
};

export type PostsProps = {
  posts: Post[];
};

const Posts = ({ posts }: PostsProps) => {
  return (
    <>
      <Head>
        <title>Posts | Ignews</title>
      </Head>
      <Base>
        <S.Wrapper>
          <S.Content>
            {posts.map(({ updatedAt, title, excerpt, slug }) => (
              <S.Link key={slug}>
                <S.Time>{updatedAt}</S.Time>
                <S.Title>{title}</S.Title>
                <S.Text>{excerpt}</S.Text>
              </S.Link>
            ))}
          </S.Content>
        </S.Wrapper>
      </Base>
    </>
  );
};

export default Posts;
