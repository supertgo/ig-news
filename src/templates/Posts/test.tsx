import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Posts from '.';

const post = {
  slug: 'monorepo-slug',
  title: 'Creating a Monorepo with Lerna & Yarn Workspaces',
  excerpt:
    'In this guide, you will learn how to create a Monorepo to manage multiple packages with a shared build, test, and release process.',
  updatedAt: '12 de março de 2021'
};

jest.mock('components/Header', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Header"></div>;
    }
  };
});

describe('<Posts />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Posts posts={[post]} />);

    expect(screen.getByText('12 de março de 2021')).toBeInTheDocument();
    expect(
      screen.getByText('Creating a Monorepo with Lerna & Yarn Workspaces')
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        'In this guide, you will learn how to create a Monorepo to manage multiple packages with a shared build, test, and release process.'
      )
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
