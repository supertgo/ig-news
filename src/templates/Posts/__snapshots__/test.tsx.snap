// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should render the heading 1`] = `
.c0 {
  max-width: 112rem;
  margin: 0 auto;
  padding: 0 3.2rem;
}

.c1 {
  max-width: 72rem;
  margin: 8rem auto 0;
}

.c2 {
  display: block;
}

.c2 + a {
  margin-top: 3.2rem;
  padding-top: 3.2rem;
  border-top: 1px solid #323238;
}

.c3 {
  font-size: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #A8A8B3;
}

.c4 {
  display: block;
  font-size: 2.4rem;
  margin-top: 3.2rem;
  line-height: 3.2rem;
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
}

.c4:hover {
  color: #EBA417;
}

.c5 {
  color: #A8A8B3;
  margin-top: 0.8rem;
  line-height: 2.6rem;
}

<div
  class=""
>
  <div
    data-testid="Mock Header"
  />
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/posts/monorepo-slug"
      >
        <time
          class="c3"
        >
          12 de mar√ßo de 2021
        </time>
        <strong
          class="c4"
        >
          Creating a Monorepo with Lerna & Yarn Workspaces
        </strong>
        <p
          class="c5"
        >
          In this guide, you will learn how to create a Monorepo to manage multiple packages with a shared build, test, and release process.
        </p>
      </a>
    </div>
  </main>
</div>
`;
